numaprom/udf/inference.py

```diff
@@ -21,75 +20,60 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def infer(payload: Payload, artifact_dict: Dict, model_config: Dict) -> Tuple[str, str]:
-    pipeline = PrometheusPipeline(
-        model_plname=model_config["model_name"],
-        model=artifact_dict["primary_artifact"],
-        seq_len=model_config["win_size"],
-        threshold_min=model_config["threshold_min"],
-    )
-    try:
-        pipeline.load_model(
-            path_or_buf=None, model=artifact_dict["primary_artifact"], **artifact_dict["metadata"]
-        )
-        LOGGER.info("%s - Successfully loaded model to pipeline", payload.uuid)
-        arr = pipeline.infer(payload.get_processed_array())
-        df = payload.get_processed_dataframe()
-        df = pd.DataFrame(data=arr, columns=df.columns, index=df.index).reset_index()
-        payload.processedMetrics = get_metrics(df)
-        payload.status = Status.INFERRED
-        payload.std = float(np.mean(pipeline.model_ppl.err_stats["std"]))
-        payload.mean = float(np.mean(pipeline.model_ppl.err_stats["mean"]))
-        payload.threshold = float(np.mean(pipeline.model_ppl.thresholds))
-        payload.model_version = artifact_dict["model_properties"].version
-        LOGGER.info("%s - Successfully inferred payload: %s", payload.uuid, payload.to_json())
-    except Exception as ex:
-        LOGGER.exception(
-            "%s - Error while doing inference. Error:%r",
-            payload.uuid,
-            ex,
-        )
-        return "", ""
-    return "postprocess", payload.to_json()
+def _run_model(
+    payload: StreamPayload, artifact_data: ArtifactData, model_config: Dict
+) -> Tuple[str, str]:
+    model = artifact_data.artifact
+    stream_data = payload.get_streamarray()
+    streamloader = DataLoader(StreamingDataset(stream_data, model_config["win_size"]))
+
+    trainer = AutoencoderTrainer()
```

Comment:
We might need to take this too from config, to make it work with different kinds of models. We can do it later in the future.

Reply:
Yes will do it later, when we tackle config management